src/main/java/core/basesyntax/SupplyOperation.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> storage, FruitTransaction transaction) {
```

Comment:
Handling of the 'supply' operation should check the transaction quantity for validity (e.g., it should not be negative). If invalid, throw a `RuntimeException`.